website/blog/2024/08-20-problems-in-web-scraping/index.md

```diff
@@ -0,0 +1,362 @@
+
+To summarize, most likely, your main library for headless browsing will be `undetected_chromedriver`. But in some particularly challenging cases, you might need to use `botasaurus`.
+
+## 4. What about frameworks?
+
+High-level frameworks are designed to speed up and ease development by allowing us to focus on business logic, although we often pay the price in flexibility and control.
+
+So, what are the frameworks for web scraping in 2024?
+
+### [Scrapy](https://docs.scrapy.org/en/latest/)
+
+It's impossible to talk about Python web scraping frameworks without mentioning Scrapy. Scrapinghub (now Zyte) first released it in 2008. For 16 years, it has been developed as an open-source library upon which development companies built their business solutions.
+
+Talking about the advantages of `Scrapy`, you could write a separate article. But I will emphasize the two of them:
+
+- The huge amount of tutorials that have been released over the years
+- Middleware libraries are written by the community and are extending their functionality. For example, [`scrapy-playwright`](https://github.com/scrapy-plugins/scrapy-playwright).
+
+But what are the downsides?
+
+In recent years, Zyte has been focusing more on developing its own platform. `Scrapy` mostly gets fixes only.
+- Lack of development towards bypassing anti-scraping systems. You have to implement them yourself, but then, why do you need a framework?
+- `Scrapy` was originally developed with the asynchronous framework `Twisted`. Partial support for `asyncio` was added only in [`version 2.0`](https://docs.scrapy.org/en/latest/topics/asyncio.html). Looking through the source code, you may notice some workarounds that were added for this purpose.
+
+Thus, `Scrapy` is a good and proven solution for sites that are not protected against web scraping. You will need to develop and add the necessary solutions to the framework in order to bypass anti-scraping measures.
+
+### [Botasaurus](https://www.omkar.cloud/botasaurus/)
+
+A new framework for web scraping using browser automation, built on [`botasaurus-driver`](https://github.com/omkarcloud/botasaurus-driver). The initial commit was made on May 9, 2023.
+
+Let's start with its advantages:
+
+- Allows you to bypass any Claudflare protection as well as many others using `botasaurus-driver`.
+- Good documentation for a quick start
+
+Downsides include:
+
+- Browser automation only, not intended for HTTP clients.
+- Tight coupling with `botasaurus-driver`; you can't easily replace it with something better if it comes out in the future.
+- No asynchrony, only multithreading.
+- At the moment, it's quite raw and still requires fixes for stable operation.
+- There are very few training materials available at the moment.
+
+This is a good framework for quickly building a web scraper based on browser automation. It lacks flexibility and support for HTTP clients, which is crutias for users like me.
+
+### [Crawlee for Python](https://www.crawlee.dev/python/)
```

Comment:
yeah, this post won't be just posted on Crawlee Blog, it will be published on Dev.to as well, also this kind of blog attracts new devs that won't know about Crawlee, they just come to read this blog and then might discover crawlee. 

so my answer is yes, it is kind of important :)

Reply:
I get it. It is just an honest review of a crawlee for python developer/contributor. It does not say anything bad, though.

@mnmkng wdyt about this?

Reply:
I think it's alright. I agree with @barjin's points, but I think the commentary is well balanced and does not say anything bad about Crawlee for Python that wouldn't be true 😅

I believe developers will appreciate the honesty. It could also help manage expectations (underpromise, overdeliver), and also drive interest in early adopters who like new libraries where they can directly contribute to their development and help shape their future.

Btw, we add notes like this to community contributed blogs:
![CleanShot 2024-08-21 at 09 57 33@2x](https://github.com/user-attachments/assets/1a7c5938-20a3-49d9-ad37-e0b0a95d2afa)

@souravjain540 Maybe we could add a similar note, or an "editor's note", saying something like "Crawlee team: Yeah, for sure we will keep improving Crawlee for Python for years to come." Or smth like that.